src/platform/platform_linux.cpp

```diff
@@ -42,23 +44,52 @@ mp::ProcessFactory* process_factory()
     }
     return static_process_factory.get();
 }
+
+QString get_driver_str()
+{
+    auto driver = qgetenv(mp::driver_env_var);
+    return driver.isEmpty() ? mp::Settings::instance().get(mp::driver_key) : driver.toLower();
+}
+
 } // namespace
 
 std::string mp::platform::default_server_address()
 {
     return {"unix:/run/multipass_socket"};
 }
 
-mp::VirtualMachineFactory::UPtr mp::platform::vm_backend(const mp::Path& data_dir)
+QString mp::platform::default_driver()
 {
-    auto driver = qgetenv("MULTIPASS_VM_DRIVER");
+    return QStringLiteral("qemu");
+}
 
-    if (driver.isEmpty() || driver == "QEMU")
+QString mp::platform::daemon_config_home() // temporary
+{
+    auto ret = QString{qgetenv("DAEMON_CONFIG_HOME")};
+    if (ret.isEmpty())
+        ret = QStringLiteral("/root/.config");
+
+    ret += QStringLiteral("/%1").arg(mp::daemon_name);
+    return ret;
+}
+
+mp::VirtualMachineFactory::UPtr mp::platform::vm_backend(const mp::Path& data_dir)
+{
+    const auto& driver = get_driver_str();
+    if (driver == QStringLiteral("qemu"))
         return std::make_unique<QemuVirtualMachineFactory>(::process_factory(), data_dir);
-    else if (driver == "LIBVIRT")
+    else if (driver == QStringLiteral("libvirt"))
         return std::make_unique<LibVirtVirtualMachineFactory>(::process_factory(), data_dir);
+    else
+    {
+        std::string msg;
+        if (driver == QStringLiteral("hyperkit") || driver == QStringLiteral("hyper-v"))
+            msg = "Unsupported virtualization driver (on this platform): {}";
+        else
+            msg = "Unsupported virtualization driver: {}";
```

Comment:
This really is protecting against someone messing with the config file manually, isn't it? I'd just throw the same error regardless of platform.

Reply:
It was just trying to help the user in case he was used to those drivers elsewhere. But yes, it is probably best to remove, I will do that.